stix2/v20/sdo.py

```diff
@@ -358,7 +358,7 @@ def __init__(self, **kwargs):
                         return
                     raise e
 
-        stix2._register_type(_Custom)
+        stix2._register_type(_Custom, version="2.0")
```

Comment:
Part of me wonders if custom objects should be version-specific. Do we envision things about how custom objects work changing between 2.0 and 2.1?

Reply:
What I had in mind is to explicitly register Custom objects for each STIX version in order to be registered. I think the developers need to be aware that properties and requirements change between version. So their objects might behave differently across STIX 2.X versions as well.

If we register Custom objects for all versions at once, for example new properties might be automatically added, their behavior might change based on changing requirements.

Also, all dictionary mappings to track internal and custom objects need to updated.

If you think a register for all versions is ok. We could have a higher level constructor that appropriately delegates the call to the correct Custom decorator and register method.

Reply:
Yeah, the common properties may change between versions. Also, a user might create a custom object with an `__init__()` that makes use of one of the new common properties. What do we do if that gets registered with an older version of the spec that doesn't have that property?

That's one reason to go for version-specific registering.

Reply:
@gtback :  I think one thing we can absolutely predict are substantive changes in point releases as all of the objects we "punted on" get defined/refined in 2.1, 2.2, ... 

So the decisions here on strategy should presume fairly disruptive changes to these libraries.  Most of of out here (self included) are not going to understand all of the underlying nuances of _ .vs. __ .vs. ___ et al.

So to the extent you can maintain an abstraction layer to protect "black box" users down the road it will be much appreciated.

Reply:
Thanks, @packet-rat. I agree with the conversation here that it's better to be explicit and require that custom types be registered explicitly with each version of STIX. This would also allow people to support (for example) the 2.1-only Location SDO in STIX 2.0 as a Custom Object, without conflicting implementations (I think).

My original question was whether something might be a "valid" 2.0 Custom Object, but no longer be valid in STIX 2.1 (or vice versa). As far as I can tell, there won't be any changes in 2.1, but who knows in 2.2 or later.